src/app/frontend/common/components/allocationchart/component.ts

```diff
@@ -12,158 +12,150 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { Component, Input, OnInit } from '@angular/core';
-import * as d3 from 'd3';
-import { Selection } from 'd3';
-import * as nv from 'nvd3';
-import { Nvd3Element } from 'nvd3';
-
-interface PieChart {
-  data: PieChartData;
-}
+import { AfterViewInit, Component, Input } from '@angular/core';
+import { generate, ChartAPI } from 'c3';
+import { BaseType, select, Selection } from 'd3';
 
 interface PieChartData {
+  key?: string;
   value: number;
+  color?: string;
 }
 
+type ChartType = 'pie' | 'donut';
+
 @Component({
   selector: 'kd-allocation-chart',
   templateUrl: './template.html',
 })
-export class AllocationChartComponent implements OnInit {
+export class AllocationChartComponent implements AfterViewInit {
   @Input() data: PieChartData[];
   @Input() colorPalette: string[];
   @Input() outerPercent: number;
   @Input() outerColor: string;
   @Input() innerPercent: number;
   @Input() innerColor: string;
-  @Input() ratio = 0.67;
+  @Input() type: ChartType = 'donut';
   @Input() enableTooltips = false;
-  @Input() size: number;
+  @Input() size = 280;
   @Input() id: string;
 
-  ngOnInit(): void {
-    this.generateGraph_();
+  allocated = new Set();
+
+  ngAfterViewInit(): void {
```

Comment:
Just curious, why can't we use `ngOnInit()` here?

Reply:
@ajatprabha It wasn't displaying the chart with `ngOnInit()`.